File: internal-frontend/src/app/dto/GatewayDto.ts
import { BatterySystemDto } from './BatterySystemDto';
import {ConsumerGroupDto} from "./ConsumerGroupDto";
import {Dto} from "./Dto";

export class GatewayDto extends Dto {
  serialNumber: string = '';
  systems: BatterySystemDto[] = [];
  groups: ConsumerGroupDto[] = [];
}

File: StorageApi/src/main/kotlin/org/vpp/storageApi/gateway/GatewayDto.kt
package org.vpp.storageApi.gateway

import kotlinx.serialization.Serializable
import org.vpp.storageApi.batterySytem.BatterySystemDto
import org.vpp.storageApi.consumerGroup.ConsumerGroupDto
import org.vpp.storageApi.dto.Dto

@Serializable
class GatewayDto : Dto() {
    lateinit var serialNumber: String
    lateinit var groups: List<ConsumerGroupDto>
    lateinit var systems: List<BatterySystemDto>
}
